core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala

```diff
@@ -103,6 +103,7 @@ private[spark] class MesosCoarseGrainedSchedulerBackend(
   private val stateLock = new ReentrantLock
 
   val extraCoresPerExecutor = conf.getInt("spark.mesos.extra.cores", 0)
+  val maxGpus = conf.getInt("spark.mesos.gpus.max", 0)
```

Comment:
Which sounds sensible to me since GPU is not usually required to run your Spark job. And also cores.max is an aggregate max, where gpu.max as the current patch is a per node max. I think I will change this into how cores.max work, but default to 0.